src/app/api/transaction/current/route.ts

```diff
@@ -0,0 +1,73 @@
+import { NextResponse } from "next/server";
+import { headers } from "next/headers";
+
+export async function GET(request: Request) {
+    try {
+        // Get authorization header
+        const headersList = headers();
+        const authorization = headersList.get("authorization");
+        
+        if (!authorization) {
+            return NextResponse.json(
+                { 
+                    error: "Unauthorized", 
+                    message: "Authorization header is required" 
+                },
+                { status: 401 }
+            );
+        }
+
+        // Extract token from "Bearer <token>" format
+        const token = authorization.startsWith("Bearer ") 
+            ? authorization.slice(7) 
+            : authorization;
```

Comment:
This is repeated in like 5 files, move them into a one file and call that from everywhere